lessons/views.py

```diff
@@ -347,9 +409,86 @@ def pay_invoice(request, reference):
             if invoice.unpaid == 0:
                 messages.add_message(request, messages.SUCCESS, f"Lesson {invoice.reference} has been fully paid!")
             else:
-                messages.add_message(request, messages.SUCCESS, f"The lesson {invoice.reference} has been overpaid by £{-invoice.unpaid}")
+                messages.add_message(request, messages.SUCCESS,
+                                     f"The lesson {invoice.reference} has been overpaid by £{-invoice.unpaid}")
             return redirect('student_invoices', invoice.student.id)
         else:
-            messages.add_message(request, messages.SUCCESS, f"A payment of £{paid} has been recorded. £{invoice.unpaid} is left to be paid")
-        
+            messages.add_message(request, messages.SUCCESS,
+                                 f"A payment of £{paid} has been recorded. £{invoice.unpaid} is left to be paid")
+
     return render(request, 'pay_invoice.html', {'invoice': invoice})
+
+
+"""
+A page to list all of the terms added to the system
+"""
+
+
+@login_required
+@permitted_groups(['admin', 'director'])
+def list_terms(request):
+    context = {"dataset": Term.objects.all()}
+    return render(request, "terms_list.html", context)
+
+
+"""
+A page to list all of the terms added to the system
+"""
+
+
+@login_required
+@permitted_groups(['admin', 'director'])
+def add_term(request):
+    context = {}
+    form = TermForm(request.POST or None)
+    if form.is_valid():
+        form.save()
+        return HttpResponseRedirect("/terms/")
+    context['form'] = form
+    return render(request, "terms_add.html", context)
+
+
+"""
+A page to list all of the terms added to the system
+"""
+
+
+@login_required
+@permitted_groups(['admin', 'director'])
+def edit_term(request, id):
+    context = {}
+    # fetch the object related to passed id
+    obj = get_object_or_404(Term, id=id)
+    # pass the object as instance in form
+    form = TermForm(request.POST or None, instance=obj)
+
+    # save the data from the form and
+    # redirect to detail_view
+    if form.is_valid():
+        form.save()
+        return HttpResponseRedirect("/terms/")
+
+    # add form dictionary to context
+    context["form"] = form
+
+    return render(request, "terms_add.html", context)
```

Comment:
Here, you should make a partial template for 'terms_add.html' then use it to make a template for adding a new term and another template for editing a term. Because now when an admin attempts to edit a term, the UI suggests that they are creating a new term